backend/pkg/api/instances.go

```diff
@@ -631,3 +639,110 @@ func (api *API) instanceStatusHistoryQuery(instanceID, appID, groupID string, li
 		Order(goqu.C("created_ts").Desc()).
 		Limit(uint(limit))
 }
+
+// instanceFactQuery returns a SelectDataset prepared to return all instances
+// that have been checked in during a given duration from a given time.
+func (api *API) instanceFactQuery(t *time.Time, duration *time.Duration) *goqu.SelectDataset {
+	if t == nil {
+		now := time.Now()
+		t = &now
+	}
+
+	if duration == nil {
+		d := time.Hour
+		duration = &d
+	}
+
+	// Convert duration to string
+	interval := fmt.Sprintf("%d milliseconds", int(duration.Milliseconds()))
+
+	return goqu.From("instance_application").
+		Select(
+			goqu.C("last_check_for_updates").As("timestamp"),
```

Comment:
Wait, that looks wrong, when we use `last_check_for_updates` we get multiple InstanceFacts entries for a version but we actually want to sum up all instances to get a single entry for how many instances are on that version.
What we should store in the final entry is `t` for the current time where we calculated the InstanceFact.

Reply:
I think it's still not correct because you group by `last_check_for_updates` and this creates an instanceFact entry per instance instead of summing them up.

Reply:
Did you revert it for debugging? The query still uses `GroupBy(goqu.L("last_check_for_updates")` which I think is not what we want and `goqu.C("last_check_for_updates").As("timestamp"),` is not correct either, we should use the given time from `timestamp` as timestamp for the generated entry.

Reply:
I think with the above it would be
```suggestion
			timestamp.As("timestamp"),
```
and the plain SQL query text should look like `select '$timestamp' as timestamp …` where `$timestamp` is the Go variable t in the right format (it's a constant, all returned rows will have the same value).